feature: first attempt at task table

tguar opened the pull request:
Looking to create a task table that can support the following:

1. a task has an id
2. a task has a title
3. a task has weight
4. a task can repeat every x days 
5. a task can repeat between floor <= x <= ceiling
6. a task can be deferred

tguar committed:
feature: first attempt

tguar commented:
I think UserId should be a foreign key, right now it's commented out because I didn't know how to properly link it

joewinnn commented on src/models/tasks.model.ts:
```diff
@@ -0,0 +1,61 @@
+
+  @AllowNull(false)
+  @Column(DataType.DECIMAL(1, 2))
+  UserWeight: number;
```
why are some fields uppercase but others lowercase? ex) taskId, title, and UserWeight?

tguar replied:
Because I didn't proofread this before I pushed haha. Let me fix that

tguar replied:
Changed, I'll let you resolve

joewinnn replied:
Not given an option to resolve.  Maybe @jin0s add me as collaborator?

jin0s replied:
my bad, I added your email instead of your github account

joewinnn commented:
Maybe I still don't understand how you're implementing the "repeating every x days"

I see fields for ceiling and floor, but where's the "x" field?

tguar commented:
> Maybe I still don't understand how you're implementing the "repeating every x days"
> 
> I see fields for ceiling and floor, but where's the "x" field?

I don't think x is defined by a value. x is just a task that falls between the two boundaries if its a range, else if its repeat every x days the ceiling is null and the floor is the value it repeats on

tguar committed:
chore: fix casing

tguar committed:
chore: fix casing

tguar committed:
feature: disallow null  RepeatCeiling

tguar committed:
feature: generate task table

jin0s commented:
> > Maybe I still don't understand how you're implementing the "repeating every x days"
> > I see fields for ceiling and floor, but where's the "x" field?
> 
> I don't think x is defined by a value. x is just a task that falls between the two boundaries if its a range, else if its repeat every x days the ceiling is null and the floor is the value it repeats on

@joewinnn Tom and I discussed it briefly, if you want "repeating every x days" this can be achieve by setting the repeating floor and ceiling numbers to the same value x. That way we get both benefits of repeating it on a specific number of days and also repeating within a range of days.

tguar committed:
feature: modify types and names

tguar committed:
chore: change comments

tguar committed:
feature: add basic task pathway

tguar committed:
feature: getTaskById working

jin0s committed:
added user.email and user.userId to jwt token returned from auth/login

jin0s committed:
CRUD for Tasks
- Create
- Get All
- Get All By Task Id
- Delete By Task Id
- Update By Task Id

See ./src/tests/manual-tests/tasks.http for examples

jin0s committed:
Move test folder out of src, added tests for Tasks
- Tested and ready to be merged to main

jin0s committed:
Merge branch 'main' of github.com:jin0s/priority-tasks into task-table
